### PR TITLE
Adding initial styled components and serverside styles

### PR DESCRIPTION
**Pull request by donovanh**
Adds styled components and an initial (empty) theme.

### CODE REVIEW

**Comment by donovanh** [comment]
Got the ThemeProvider working server side. Just need to work out how to inject the global styling and we'll be go.

**Review by IvanJov — APPROVED** [review]
👏

**Comment by donovanh** [comment]
I've set up some global CSS rules as a string to be reusable on client and server:

```
const { ServerStyleSheet, injectGlobal } = require('styled-components')
const globalStyleRules = require('../../client/src/theme/globalStyleRules')

injectGlobal`${globalStyleRules}`
```

Should be working now on both client and server. And Storybook.

File: client/.storybook/config.js [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
 
+globalStyles()
+
+addDecorator(story => <ThemeProvider>{story()}</ThemeProvider>)
+
+// automatically import all files ending in *.stories.js
+const req = require.context('../src/stories', true, /\.js$/)
 function loadStories() {
-  require('../src/stories');
+  req.keys().forEach(filename => req(filename))
```
Comment (feugy) [inline client/.storybook/config.js:13]: :+1:

File: client/src/index.js [hunk 2]
```diff
@@ -1,21 +1,21 @@
 import React from 'react'
 import ReactDOM from 'react-dom'
 import './index.css'
```
Comment (feugy) [inline client/src/index.js]: We should get rid of this one, shoudn't we?

File: client/src/index.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ if (typeof window !== 'undefined') {
       )
     })
   }
-  
+
```
Comment (feugy) [inline client/src/index.js:28]: Silly me: I'll cehck why linter didn't removed them.

**Comment by donovanh** [comment]
(The above is found here :https://github.com/donovanh/react-server-demo/pull/7/files#diff-e3b125de71c23b14602016da3269123d)

File: server/middleware/renderer.js [hunk 4]
```diff
@@ -7,6 +7,11 @@ const { App } = require('../../client/build/static/js')
 const path = require('path')
 const fs = require('fs')
 
+const { ServerStyleSheet, injectGlobal } = require('styled-components')
+const globalStyleRules = require('../../client/src/theme/globalStyleRules')
+
+injectGlobal`${globalStyleRules}`
```
Comment (feugy) [inline server/middleware/renderer.js]: Should be in the exported function, unless there's a good reason to run it when requiring the file.

I'm always cautious with such side effects...

**Review by feugy — COMMENTED** [review]
Very educative!
I cannot wait to try it.

Some comments to double check

File: client/src/index.js (hunk 2 above)
Comment (donovanh) [reply client/src/index.js]: :+1:

**Pull request merged**